tests/python_package_test/test_arrow.py

```diff
@@ -454,3 +454,33 @@ def test_arrow_feature_name_manual():
     )
     booster = lgb.train({"num_leaves": 7}, dataset, num_boost_round=5)
     assert booster.feature_name() == ["c", "d"]
+
+
+def test_dataset_construction_from_pa_table_without_cffi_raises_informative_error(missing_module_cffi):
+    with pytest.raises(
+        lgb.basic.LightGBMError, match="Cannot init Dataset from Arrow without 'pyarrow' and 'cffi' installed."
+    ):
+        lgb.Dataset(
+            generate_dummy_arrow_table(),
+            label=pa.array([0, 1, 0, 0, 1]),
+            params=dummy_dataset_params(),
+        ).construct()
+
+
+def test_predicting_from_pa_table_without_cffi_raises_informative_error(missing_module_cffi):
+    data = generate_random_arrow_table(num_columns=3, num_datapoints=1_000, seed=42)
+    labels = generate_random_arrow_array(num_datapoints=data.shape[0], seed=42)
+    dataset = lgb.Dataset(
+        data.to_pandas(),
+        label=labels.to_pandas(),
+    )
+
+    with pytest.raises(
+        lgb.basic.LightGBMError, match="Cannot predict from Arrow without 'pyarrow' and 'cffi' installed."
+    ):
+        bst = lgb.train(
```

Comment:
I think `lgb.train()` part should be outside of the `with` block because we expect only `predict()` should fail.

Reply:
You are absolutely right, thank you!

Fixed in https://github.com/microsoft/LightGBM/pull/6785/commits/7396613590c06e03876612df30f09e9e0ee2e44a